src/components/CanvasWrapper.tsx

```diff
@@ -178,27 +179,22 @@ export default function CanvasWrapper(inputProps: CanvasWrapperProps): ReactElem
   const lastMousePositionPx = useRef(new Vector2(0, 0));
   const theme = useContext(AppThemeContext);
 
-  const [showMissingFileIcon, setShowMissingFileIcon] = useState(false);
+  const isMissingFile = !frameLoadResult.frameLoaded || !frameLoadResult.backdropLoaded;
 
   // CANVAS PROPERTIES /////////////////////////////////////////////////
 
-  const onFrameChangedCallback = useCallback(
-    (isMissing: boolean) => {
-      setShowMissingFileIcon(isMissing);
-      if (props.showAlert && isMissing) {
-        props.showAlert({
-          type: "warning",
-          message: "Warning: One or more frames failed to load.",
-          description: LoadTroubleshooting.CHECK_FILE_OR_NETWORK,
-          showDoNotShowAgainCheckbox: true,
-          closable: true,
-        });
-      }
-    },
-    [props.showAlert, setShowMissingFileIcon, canv]
-  );
-
-  canv.setOnFrameChangeCallback(onFrameChangedCallback);
+  // Show warning if files are missing
+  useMemo(() => {
+    if (isMissingFile && props.showAlert) {
+      props.showAlert({
+        type: "warning",
+        message: "Warning: One or more frames or backdrops failed to load.",
+        description: LoadTroubleshooting.CHECK_FILE_OR_NETWORK,
+        showDoNotShowAgainCheckbox: true,
+        closable: true,
+      });
+    }
+  }, [frameLoadResult, isMissingFile, props.showAlert]);
```

Comment:
Technically this warning doesn't need to exist here... I could move it to `Viewer.tsx`, but I like having it here since CanvasWrapper also shows an onscreen icon for missing files too.